pkg/generate/app/sourcelookup.go

```diff
@@ -44,45 +44,32 @@ type SourceRepository struct {
 // NewSourceRepository creates a reference to a local or remote source code repository from
 // a URL or path.
 func NewSourceRepository(s string) (*SourceRepository, error) {
-	var location *url.URL
-	switch {
-	case strings.HasPrefix(s, "git@"):
-		base := "git://" + strings.TrimPrefix(s, "git@")
-		url, err := url.Parse(base)
-		if err != nil {
-			return nil, err
-		}
-		location = url
+	uri, err := url.Parse(s)
```

Comment:
----- Original Message -----
> 
> 
> ----- Original Message -----
> > > @@ -44,45 +44,32 @@ type SourceRepository struct {
> > >  // NewSourceRepository creates a reference to a local or remote source
> > >  code repository from
> > >  // a URL or path.
> > >  func NewSourceRepository(s string) (*SourceRepository, error) {
> > > -	var location *url.URL
> > > -	switch {
> > > -	case strings.HasPrefix(s, "git@"):
> > > -		base := "git://" + strings.TrimPrefix(s, "git@")
> > > -		url, err := url.Parse(base)
> > > -		if err != nil {
> > > -			return nil, err
> > > -		}
> > > -		location = url
> > > +	uri, err := url.Parse(s)
> > 
> > Although I initially focus entirely on ssh keys, after your suggestion I
> > reworked my idea. Currently I've defined secret to be used with git auth to
> > look like this:
> > ```
> > {
> >   "apiVersion": "v1beta3",
> >   "kind": "Secret",
> >   "metadata": {
> >     "name": "mygitsecret"
> >   },
> >   "data": {
> >     "sshsecret": "ZnVCWX...."
> >   }
> > }
> > ```
> > Since each method requires two things actually, the method name to be used
> > and the actual secret. I was thinking to hide that in above `data` field.
> > Where the key would be the method to use (`sshsecret`, `basicsecrect` etc.)
> > and the value would be the actual data to be used. With all that, we'd
> > define the precedence of those methods, just in case somebody would define
> > all of them in one secret, I was thinking ssh keys, certs/tokens and lastly
> > basic auth.
> > I can add above description to the follow up
> > [card](https://trello.com/c/6EhPEbM4/569-add-other-auth-methods-for-accessing-private-repos).
> > I'll try to catch you on IRC today to discuss that idea.
> > 
> 
> For naming, trying to pick a name that would roughly resemble what an end
> user sees is good - for instance, "ssh" might be better than "sshsecret"
> because we know it's secret, and for the others "basic" and "token" will
> work fine.  I am a bit worried about defining a schema here - we maybe
> should set the secret type to "openshift.io/Build" so it's clear this
> follows a certain scheme (@liggitt, opinions?)

Discussion from IRC:

1. The user should provide a secret that has "data" keys we recognize
2. The builder should report what keys it's trying to use and in what order
3. If the builder gets a secret but no keys are recognized it should print a clear warning

Possible up front types

"ssh-privatekey" -> openssh compatible private key (identity key)
"user-password" -> <user>:<password>
"bearer-token" -> <token>

Future types:

"client-certificate" -> PEM encoded client certificate

Example warning:

WARNING: The provided secret 'foo' did not have any of the supported keys 'ssh-privatekey', 'user-password', or 'bearer-token'

Example message:

Using 'ssh-privatekey' from secret 'foo'
Using 'ssh-privatekey', 'user-password' from secret 'foo'